Lib/logging/__init__.py

```diff
@@ -413,38 +416,54 @@ def makeLogRecord(dict):
     rv.__dict__.update(dict)
     return rv
 
+
 #---------------------------------------------------------------------------
 #   Formatter classes and functions
 #---------------------------------------------------------------------------
-
 class PercentStyle(object):
 
     default_format = '%(message)s'
     asctime_format = '%(asctime)s'
     asctime_search = '%(asctime)'
+    validation_pattern = r'(\%\([^%()]+\)[#|0|\-|\s|+]?[\d]*[d|i|o|u|x|X|e|E|f|E|g|G|C|r|s|a|%]{1})'
 
     def __init__(self, fmt):
         self._fmt = fmt or self.default_format
 
     def usesTime(self):
         return self._fmt.find(self.asctime_search) >= 0
 
-    def format(self, record):
+    def validate(self):
+        if not re.search(self.validation_pattern, self._fmt):
+            raise ValueError('Invalid format %(fmt)s for %(style)s style'
+                             % {"fmt": self._fmt, "style": self.default_format[0]})
+
+    def _format(self, record):
         return self._fmt % record.__dict__
 
+    def format(self, record):
+        try:
+            return self._format(record)
+        except KeyError as e:
+            raise ValueError("Invalid formatting field %s" % e)
```

Comment:
I suggested earlier `raise ValueError('Formatting field not found in record: %s' % e)`, as I thought that more informative. Do you not agree? "Invalid formatting field" suggests it might be improperly formatted, but the real problem is that is not present in the record, and not how it's formatted (which your wording might be misinterpreted to mean).